### PR TITLE
Enables backwards compatibility with include paths prefixed with "ndk/"

### PR DESCRIPTION
**Pull request by ahumesky**
Also adds tests, and a note about using https://github.com/google/cpu_features instead of the cpu_features target defined by android_ndk_repository.

### CODE REVIEW

File: rules.bzl [hunk 1]
```diff
@@ -106,13 +106,16 @@ def _create_symlinks(ctx, ndk_path, clang_directory, sysroot_directory):
 
     ctx.symlink(ndk_path + "sources", "sources")
 
+    # Remove this hack, see https://github.com/bazelbuild/rules_android_ndk/issues/32
```
Comment (ted-xie) [inline rules.bzl:109]: ditto

**Review by ted-xie — CHANGES_REQUESTED** [review]
just a few nits, otherwise LGTM

File: rules.bzl (hunk 1 above)
Comment (ahumesky) [reply rules.bzl]: Done

File: BUILD.ndk_root.tpl [hunk 2]
```diff
@@ -24,7 +24,15 @@ alias(
...
-    srcs = glob(["sources/android/cpufeatures/*.c"]),
-    hdrs = glob(["sources/android/cpufeatures/*.h"]),
+    srcs = glob([
+        "sources/android/cpufeatures/*.c",
+        # Remove this hack, see https://github.com/bazelbuild/rules_android_ndk/issues/32
+        "ndk/sources/android/cpufeatures/*.c",
+    ]),
+    hdrs = glob([
+        "sources/android/cpufeatures/*.h",
+        # Remove this hack, see https://github.com/bazelbuild/rules_android_ndk/issues/32
```
[the remark this answers is not in the document]
Comment (ahumesky) [reply BUILD.ndk_root.tpl]: Done

File: BUILD.ndk_root.tpl [hunk 3]
```diff
@@ -24,7 +24,15 @@ alias(
 
 cc_library(
     name = "cpufeatures",
-    srcs = glob(["sources/android/cpufeatures/*.c"]),
-    hdrs = glob(["sources/android/cpufeatures/*.h"]),
+    srcs = glob([
+        "sources/android/cpufeatures/*.c",
+        # Remove this hack, see https://github.com/bazelbuild/rules_android_ndk/issues/32
```
[the remark this answers is not in the document]
Comment (ahumesky) [reply BUILD.ndk_root.tpl]: Done

**Review by ted-xie — APPROVED** [review]

**Pull request merged**